src/jit/importer.cpp

```diff
@@ -7208,10 +7208,6 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
 
                 GenTreePtr thisPtr = impPopStack().val;
                 thisPtr            = impTransformThis(thisPtr, pConstrainedResolvedToken, callInfo->thisTransform);
-                if (compDonotInline())
-                {
-                    return callRetTyp;
-                }
```

Comment:
`impTransformThis` can fail, so don't we still need the check?

Reply:
on the line 7123 we have:
```
if (compIsForInlining())
{
    compInlineResult->NoteFatal(InlineObservation::CALLSITE_HAS_CALL_VIA_LDVIRTFTN);
    return TYP_UNDEF;
}
```
that abort all inlinings. 
It was tested in my clean up PR #13091 with unreached statements.

Reply:
Can you then add an assert that `impTransformThis` returns something non-null hre? I worry about the case where somebody removes the initial bail-out check and doesn't realize they now may need new bail-out checks lower down.

Reply:
Fixed